model/host/host_test.go

```diff
@@ -4568,37 +4580,45 @@ func TestStartingHostsByClient(t *testing.T) {
 			Provider: evergreen.ProviderNameEc2OnDemand,
 			Region:   evergreen.DefaultEC2Region,
 		}:
-			assert.Len(t, hosts, 2)
-			assert.Contains(t, hosts, startingHosts[0])
-			assert.Contains(t, hosts, startingHosts[1])
+			require.Len(t, hosts, 2)
+			compareHosts(t, hosts[0], startingHosts[0])
+			compareHosts(t, hosts[1], startingHosts[1])
 		case ClientOptions{
 			Provider: evergreen.ProviderNameDocker,
 		}:
-			assert.Len(t, hosts, 2)
-			assert.Contains(t, hosts, startingHosts[2])
-			assert.Contains(t, hosts, startingHosts[3])
+			require.Len(t, hosts, 2)
+			compareHosts(t, hosts[0], startingHosts[2])
+			compareHosts(t, hosts[1], startingHosts[3])
 		case ClientOptions{
 			Provider: evergreen.ProviderNameEc2Spot,
-			Region:   "us-west1",
+			Region:   "us-west-1",
 			Key:      "key1",
 			Secret:   "secret1",
 		}:
-			assert.Len(t, hosts, 1)
-			assert.Contains(t, hosts, startingHosts[4])
+			require.Len(t, hosts, 1)
+			compareHosts(t, hosts[0], startingHosts[4])
 		case ClientOptions{
 			Provider: evergreen.ProviderNameEc2Spot,
-			Region:   "us-west1",
+			Region:   "us-west-1",
 			Key:      "key2",
 			Secret:   "secret2",
 		}:
-			assert.Len(t, hosts, 1)
-			assert.Contains(t, hosts, startingHosts[5])
+			require.Len(t, hosts, 1)
+
+			compareHosts(t, hosts[0], startingHosts[5])
 		default:
 			assert.Fail(t, "unrecognized client options")
 		}
 	}
 }
 
+func compareHosts(t *testing.T, host1, host2 Host) {
+	assert.Equal(t, host1.Id, host2.Id)
+	assert.Equal(t, host1.Status, host2.Status)
+	assert.Equal(t, host1.Distro.Provider, host2.Distro.Provider)
+	assert.Equal(t, host1.Distro.ProviderSettingsList[0].ExportMap(), host2.Distro.ProviderSettingsList[0].ExportMap())
```

Comment:
does this really work? 

I think there's an "Equal" method on birch documents that might be useful here,

Reply:
There's an Equal method on Elements and Values but not on documents right now, and comparing documents with just the assert Equals complains about the pointers being different. This does work though!